packages/styled-components/src/models/InlineStyle.js

```diff
@@ -34,6 +34,26 @@ export default (styleSheet: StyleSheet) => {
         const declPairs = [];
         root.each(node => {
           if (node.type === 'decl') {
+            // Handle percentage and em for font-size
+            if(node.prop === 'font-size') {
+              const fontSizeUnit = node.value.match(/(em|%)$/)
+              if(fontSizeUnit) {
+                let value = parseInt(node.value.substring(0, fontSizeUnit.index), 10)
+                if(fontSizeUnit[0] === '%') value /= 100
+                // eslint-disable-next-line no-param-reassign
+                node.value = `${value}`
+                // We need to warn that the font size is specified in em
+                // eslint-disable-next-line no-param-reassign
+                unitsInformation.fontSizeInEm = true
+              }
+            }
+            // We strip the special units and mark them for later treatment
+            const extendedUnits = node.value.match(/(em|vh|vw)$/)
```

Comment:
This isn't quite right, as it's possible to have multiple units in one definition - like `padding: 10vw 10vh`

Reply:
Damn. Hard to keep in mind all the possible cases...
I will need to keep track of the properties that need to be updated after the call to css-to-react-native. Do you guys have a list of this kind of properties somewhere ?

Reply:
Ok, here is the list I can think of:
  * can receive multiple unit values : ['padding', 'margin']
  * can receive multiple values among which there may be one unit : ['font', 'border', 'border-top', 'border-bottom', 'border-left', border-right', 'flex', 'background']
The current code should be taking care correctly of other properties.

Reply:
If you pass in the viewport dimensions, you can do a regexp on each value and convert it to px values, then hand it off to css-to-react-native

E.g. `padding: 10vw` -> `padding: 48px`

If you use the callback String.replace, you can even track whether you did this replacement so you know whether the stylesheet depended on viewport dimensions

If you need some regexps, you can also find them [here](https://github.com/jacobp100/cssta/blob/8c9697e8d4f5ddd6c2ff9333af42ec62e31a6773/runtime/cssRegExp.ts)

Reply:
Ok, I think I got them all, but if I forgot some, please tell me!
It would indeed be a good way to provide the width and height of the screen and make the calculation within the script. But, I need to take care of the em anyway and this can only be done in the component...

Reply:
I think this is an exhaustive list of CSS properties that can receive one of those units:

background
background-position
background-size
border
border-bottom
border-bottom-left-radius
border-bottom-right-radius
border-bottom-width
border-image
border-image-outset
border-image-slice
border-image-width
border-left
border-left-width
border-radius
border-right
border-right-width
border-spacing
border-top
border-top-left-radius
border-top-right-radius
border-top-width
border-width
bottom
box-shadow
clip
font
font-size
font-size-adjust
height
left
letter-spacing
line-height
margin
margin-bottom
margin-left
margin-right
margin-top
max-height
max-width
min-height
min-width
outline
outline-width
padding
padding-bottom
padding-left
padding-right
padding-top
right
text-indent
top
transform
transform-origin
width
word-spacing

From: https://htmldog.com/references/css/properties/

Reply:
Great, thx! I'll add this, but I'd like to be sure that this has a chance to be merged before working more on the details.
I need some help from anyone knowing flow, and ideally a principe approval from anyone maintaining this repo...

Reply:
These are the properties that don't support other units for now:

- background
- background-position
- background-size
- border-image
- border-image-outset
- border-image-slice
- box-shadow
- clip
- outline
- transform
- transform-origin

I have no solution for transform and clip. I think that the rest can be fixed as soon as a maintainer answers me on this PR...